Artesian/Artesian.SDK/Service/Query/VersionedQuery.cs

```diff
@@ -186,6 +186,23 @@ public VersionedQuery ForMostRecent(LocalDate start, LocalDate end)
             if (end <= start)
```

Comment:
Why not make it that the LocalDate variant to call into LocalDateTime one with AtMidnight(), avoiding code duplication?